samza-core/src/main/scala/org/apache/samza/diagnostics/DiagnosticsStreamMessage.java

```diff
@@ -228,6 +239,14 @@ public Boolean getAutosizingEnabled() {
     return (Boolean) getFromMetricsMessage(GROUP_NAME_FOR_DIAGNOSTICS_MANAGER, AUTOSIZING_ENABLED_METRIC_NAME);
   }
 
+  /**
+   * This method gets the config of the job from the MetricsMessage.
+   * @return the config of the job.
+   */
+  public Config getConfig() {
+    return (Config) getFromMetricsMessage(GROUP_NAME_FOR_DIAGNOSTICS_MANAGER, CONFIG);
```

Comment:
Since we will serialize as Map<String, String> 
getFromMetricsMessage will return a Map<String, String> 
that needs to be converted to Config like this: 

new MapConfig(getFromMetricsMessage(GROUP_NAME_FOR_DIAGNOSTICS_MANAGER, CONFIG));

Reply:
Done.